public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,318 @@
+                              return (
+                                <Select
+                                  {...field}
+                                  value={fieldVal.type}
+                                  onChange={(value) => {
+                                    if (!readOnly) {
+                                      const currentValues = getValues().config.transformations[index];
+                                      let keptValsCopy = fill(Array(index + 1), {});
+                                      keptVals.forEach((keptVal, i) => (keptValsCopy[i] = keptVal));
+                                      keptValsCopy[index] = {
+                                        expression: currentValues.expression,
+                                        mapValue: currentValues.mapValue,
+                                      };
+
+                                      setKeptVals(keptValsCopy);
+
+                                      const newValueDetails = getSupportedTransTypeDetails(value.value);
+
+                                      if (newValueDetails.showExpression) {
+                                        setValue(
+                                          `config.transformations.${index}.expression`,
+                                          keptVals[index]?.expression || ''
+                                        );
+                                      } else {
+                                        setValue(`config.transformations.${index}.expression`, '');
+                                      }
+
+                                      if (newValueDetails.showMapValue) {
+                                        setValue(
+                                          `config.transformations.${index}.mapValue`,
+                                          keptVals[index]?.mapValue || ''
+                                        );
+                                      } else {
+                                        setValue(`config.transformations.${index}.mapValue`, '');
+                                      }
+
+                                      onChange(value.value);
+                                    }
+                                  }}
+                                  options={transformOptions}
+                                  width={25}
+                                  aria-label="Type"
+                                />
+                              );
+                            }}
+                            control={control}
+                            name={`config.transformations.${index}.type`}
+                            rules={{ required: { value: true, message: 'Please select a transformation type' } }}
+                          />
+                        </Field>
+                        <Field
+                          label={
+                            <Stack gap={0.5}>
+                              <Label>Field</Label>
```

Comment:
we should add an `htmlFor` here:
```suggestion
                              <Label htmlFor={`config.transformations.${fieldVal.id}.field`}>Field</Label>
```